src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c

```diff
@@ -21,69 +87,1182 @@ const ActorInit Boss_Fd2_InitVars = {
+            break;
+        }
+    }
+}
+
+void BossFd2_SpawnFireBreath(GlobalContext* globalCtx, BossFdParticle* particle, Vec3f* position, Vec3f* velocity,
+                             Vec3f* acceleration, f32 scale, s16 opacity, s16 kbAngle) {
+    s16 i;
+
+    for (i = 0; i < 180; i++, particle++) {
+        if (particle->type == 0) {
+            particle->type = 4;
+            particle->timer1 = 0;
+            particle->pos = *position;
+            particle->vel = *velocity;
+            particle->accel = *acceleration;
+            particle->pos.x -= particle->vel.x;
+            particle->pos.y -= particle->vel.y;
+            particle->pos.z -= particle->vel.z;
+            particle->unk_34 = 0.0f;
+            particle->opacity = opacity;
+            particle->unk_38 = Math_Rand_ZeroFloat(10.0f);
+            particle->timer2 = 0;
+            particle->scale = scale / 400.0f;
+            particle->kbAngle = kbAngle;
+            break;
+        }
+    }
+}
+
+void BossFd2_SpawnEmber(GlobalContext* globalCtx, BossFdParticle* particle, Vec3f* position, Vec3f* velocity,
+                        Vec3f* acceleration, f32 scale) {
+    s16 i;
+
+    for (i = 0; i < 180; i++, particle++) {
+        if (particle->type == 0) {
+            particle->type = 1;
+            particle->pos = *position;
+            particle->vel = *velocity;
+            particle->accel = *acceleration;
+            particle->scale = scale / 1000.0f;
+            particle->opacity = 0xFF;
+            particle->timer1 = (s16)Math_Rand_ZeroFloat(10.0f);
+            break;
+        }
+    }
+}
+
+void BossFd2_SpawnSkullPiece(GlobalContext* globalCtx, BossFdParticle* particle, Vec3f* position, Vec3f* velocity,
+                             Vec3f* acceleration, f32 scale) {
+    s16 i;
+
+    for (i = 0; i < 180; i++, particle++) {
+        if (particle->type == 0) {
+            particle->type = 5;
+            particle->pos = *position;
+            particle->vel = *velocity;
+            particle->accel = *acceleration;
+            particle->scale = scale / 1000.0f;
+            particle->unk_34 = Math_Rand_ZeroFloat(100.0f);
+            particle->unk_38 = Math_Rand_ZeroFloat(100.0f);
+            break;
+        }
+    }
+}
+
+void BossFd2_SpawnDust(BossFdParticle* particle, Vec3f* position, Vec3f* velocity, Vec3f* acceleration, f32 scale) {
+    s16 i;
+
+    for (i = 0; i < 180; i++, particle++) {
+        if (particle->type == 0) {
+            particle->type = 3;
+            particle->pos = *position;
+            particle->vel = *velocity;
+            particle->accel = *acceleration;
+            particle->timer2 = 0;
+            particle->scale = scale / 400.0f;
+            break;
+        }
+    }
+}
+
+void BossFd2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BossFd2* this = THIS;
+    ColliderJntSph* sp34;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, 0.0069999993f);
```

Comment:
Does `0.007f` work?

Reply:
No. There might be a plausible improperly typed multiplication that gives that, but it's off by 0x00000002, which is much harder to finagle than if it were off by 0x00000001.

Reply:
I guess `0.007f * (1.0f - 0.0000001f)` would work then but it's not exactly sane anyway